src/adapter/src/catalog/config.rs

```diff
@@ -74,6 +75,8 @@ pub struct Config<'a> {
     /// TODO(migration): delete in version v.51 (released in v0.49 + 1
     /// additional release)
     pub connection_context: Option<mz_storage_client::types::connections::ConnectionContext>,
+    /// Global connection limit and count
+    pub active_connection_count: Arc<std::sync::Mutex<ConnectionCounter>>,
```

Comment:
It's unfortunate the shared state needs to live in the catalog and its config, but that does indeed seem to be what's needed for SET and RESET to function.

Reply:
Yeah we need the global state somewhere. I assume the config is short lived so doesn't matter?